File: RecyclerView/app/src/main/java/com/bridgelabz/util/recyclerview/MainActivity.java
package com.bridgelabz.util.recyclerview;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.widget.RelativeLayout;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    public RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        RelativeLayout relativeLayout = new RelativeLayout(this);

        recyclerView = (RecyclerView) findViewById(R.id.recyclerView);

        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        relativeLayout.addView(recyclerView);

        ArrayList<studentSubjects> subjects = new ArrayList();

        studentSubjects math = new studentSubjects("Math", false, R.drawable.maths);
        subjects.add(math);

        studentSubjects science = new studentSubjects("Science", true, R.drawable.science);
        subjects.add(science);

        studentSubjects reading = new studentSubjects("Reading", false, R.drawable.reading);
        subjects.add(reading);

        studentSubjects history = new studentSubjects("History", false, R.drawable.history);
        subjects.add(history);

        studentSubjects biology = new studentSubjects("Biology", false, R.drawable.biology);
        subjects.add(biology);

        StudentAdapter studentAdapter = new StudentAdapter(this, subjects);
        recyclerView.setAdapter(studentAdapter);
    }
}

File: kotlin/foodVolley/app/src/main/java/com/bridgelabz/foodvolley/MainActivity.kt
package com.bridgelabz.foodvolley

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import com.android.volley.Request
import com.android.volley.Response
import com.android.volley.toolbox.JsonArrayRequest
import com.android.volley.toolbox.Volley
import kotlinx.android.synthetic.main.activity_main.*
import org.json.JSONArray
import org.json.JSONObject
import java.lang.reflect.Method

class MainActivity : AppCompatActivity() {

    val arrayData = ArrayList<dataModel>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        RecyclerViewFood.layoutManager = LinearLayoutManager(this)
        RecyclerViewFood.setHasFixedSize(true)

        var jsonReqQueue = Volley.newRequestQueue(this)

        var jsonReq = JsonArrayRequest(Request.Method.GET, "https://choco-lava.herokuapp.com/borrow/getfood", null,
                Response.Listener {
                    response ->  getAllData(response)
                }, Response.ErrorListener {
            error -> print(error)
        })

        jsonReqQueue.add(jsonReq)
    }

    private fun getAllData(response: JSONArray) {

        for( i in 0..response.length()-1 ){

            var jsonObl : JSONObject = response.get(i) as JSONObject
            val data = dataModel(jsonObl.getString("name"), jsonObl.getString("image"))
            arrayData.add(data)

        }
        RecyclerViewFood.adapter = myRecyclerAdapter(this, arrayData)
    }
}
